src/components/my-url-input.ts

```diff
@@ -44,59 +67,86 @@ export class MyUrlInputMain extends LitElement {
     }
   }
 
+  private _handleAdditional() {
+    this.values = [...this.values || [], ""]
+  }
+
+  private _handleRemove(event) {
+    const remove = (event.currentTarget as TextField).value
+    const filtered = this.values?.filter(value => value !== remove)
+    this.values = filtered?.length ? filtered : undefined
+  }
+
   private _handleSave() {
-    const inputEl = this.shadowRoot!.querySelector("mwc-textfield")!;
-    let value = inputEl.value || "";
-    this._error = undefined;
+    let values = [...this.values || []]
 
-    if (value === "") {
-      value = DEFAULT_HASS_URL;
-    }
+    for (const inputEl of this.shadowRoot!.querySelectorAll("mwc-textfield")) {
+      let value = inputEl.value || "";
+      this._error = undefined;
 
-    if (value.indexOf("://") === -1) {
-      this._textfield.setCustomValidity(
-        "Please enter your full URL, including the protocol part (https://)."
-      );
-      this._textfield.reportValidity();
-      return;
+      if (value === "") {
+        value = DEFAULT_HASS_URL;
+      }
+  
+      if (value.indexOf("://") === -1) {
+        inputEl.setCustomValidity(
+          "Please enter your full URL, including the protocol part (https://)."
+        );
+        inputEl.reportValidity();
+        return;
+      }
+  
+      let urlObj: URL;
+      try {
+        urlObj = new URL(value);
+      } catch (err) {
+        inputEl.setCustomValidity("Invalid URL");
+        inputEl.reportValidity();
+        return;
+      }
+      values.push(`${urlObj.protocol}//${urlObj.host}`);
     }
 
-    let urlObj: URL;
+    values = Array.from(new Set(values)).filter(url => url != "");
+    
     try {
-      urlObj = new URL(value);
+      window.localStorage.setItem(STORAGE_KEY, JSON.stringify(values));
     } catch (err) {
-      this._textfield.setCustomValidity("Invalid URL");
-      this._textfield.reportValidity();
+      this._error = "Failed to store your URLs!";
       return;
     }
-    const url = `${urlObj.protocol}//${urlObj.host}`;
-    try {
-      window.localStorage.setItem(HASS_URL, url);
-    } catch (err) {
-      this._error = "Failed to store your URL!";
-      return;
-    }
-    fireEvent(this, "value-changed", { value: url });
+    fireEvent(this, "value-changed", { value: values });
   }
 
   static get styles(): CSSResult {
     return css`
       :host {
         display: block;
       }
-      div {
-        display: flex;
-        justify-content: space-between;
-        align-items: center;
-      }
       .error {
         color: #db4437;
         font-weight: bold;
       }
+      .instance-input, .buttons {
+        display: flex;
+        justify-content: space-between;
+      }
+      .buttons {
+        margin-right: 64px;
```

Comment:
Without this margin the save button lines up with the remove button. That's not bad either:

![image](https://user-images.githubusercontent.com/1444314/195939295-9d1a5e10-93f2-4d28-974b-10832fde64c2.png)

Reply:
And it's then more material-y, as it's in the corner.